common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java

```diff
@@ -211,49 +249,16 @@ public String decodeState(final String encodedState) {
         return new String(stateBytes, Charset.defaultCharset());
     }
 
+    @Override
+    public Uri getAuthorizationRequestAsHttpRequest() throws UnsupportedEncodingException {
+        String queryStringParameters = ObjectMapper.serializeObjectToFormUrlEncoded(this);
+        Uri.Builder uriBuilder = Uri.parse(getAuthorizationEndpoint()).buildUpon()
```

Comment:
When i ran this code yesterday.... there was a problem here since the "?" denoting the beginning of the query string is not added.  

I modifed to this:

HashMap<String, String> params = ObjectMapper.serializeObjectHashMap(this);
        Uri.Builder uriBuilder = Uri.parse(getAuthorizationEndpoint()).buildUpon();

        Iterator it = params.entrySet().iterator();

        while(it.hasNext()){
            Map.Entry<String, String> param = (Map.Entry<String, String>)it.next();
            uriBuilder.appendQueryParameter(param.getKey(), param.getValue());
        }

        return uriBuilder.build();

Reply:
Thanks a lot, Shane. I will work on it.